compiler/modules/nback/number.bal

```diff
@@ -174,39 +183,51 @@ function buildConvertToFloat(llvm:Builder builder, Scaffold scaffold, bir:Conver
         buildConvertIntToFloat(builder, scaffold, val, insn);
         return;
     }
-    else if repr.base != BASE_REPR_TAGGED {
-        return scaffold.unimplementedErr("convert form decimal to float", insn.pos);
+    else if repr == REPR_DECIMAL {
+        buildConvertDecimalToFloat(builder, scaffold, val, insn);
+        return;
     }
     // convert to float form tagged pointer
 
     // number part of semType must be some *non-empty* combination of
     // (some or all of) int, float and decimal
-    t:SemType semType = insn.operand.semType;
     llvm:PointerValue tagged = <llvm:PointerValue>val;
     llvm:BasicBlock joinBlock = scaffold.addBasicBlock();
-
-    // semType must contain int or decimal. Since we don't have decimal yet in subset 6,
-    // it must contain int. In the future, below section is only needed conditionally.
-    llvm:Value hasType = buildHasTag(builder, tagged, TAG_INT);
-    llvm:BasicBlock hasIntBlock = scaffold.addBasicBlock();
-    llvm:BasicBlock noIntBlock = scaffold.addBasicBlock();
-    builder.condBr(hasType, hasIntBlock, noIntBlock);
-    builder.positionAtEnd(hasIntBlock);
-    buildConvertIntToFloat(builder, scaffold, buildUntagInt(builder, scaffold, tagged), insn);
-    builder.br(joinBlock);
-
-    builder.positionAtEnd(<llvm:BasicBlock>noIntBlock);
-    if !t:isSubtypeSimple(semType, t:INT) {
+    buildConvertTaggedToFloat(builder, scaffold, tagged, insn, TAG_INT, joinBlock);
```

Comment:
Better to handle the case where we are going from tagged ptr to tagged ptr with a runtime function.

Reply:
Created a runtime function called `_bal_tagged_to_tagged_float`. My plan is to create another function called `_bal_tagged_to_tagged_int`, under `ConvertToIntInsn`